Add git hooks

artemkovalyov opened the pull request:
## What Has Changed?

To have less load on our pipeline and ensure styles I created pre-push hooks to run prettier (with fixes) and vale on what's pushed to the PR. This is standard practice and lets us ensure DoDs and conventions in an automated way.

Make sure you have vale installed. That should be rather easy: https://docs.errata.ai/vale-server/install/


## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on its own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [x] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] You verified all new and changed links still work (changing the `id` or name of a file can break links)

florian-richter commented:
I don't think there is a hook included in this PR, probably due to the `.git` directory being excluded. At least when I checked out this branch, I could still commit something that had a vale error in it.

artemkovalyov commented:
@florian-richter, I simplified the hooks by using Husky. You just have to run npm install:

```bash
± |add-git-hooks ✓| → npm install
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.

> sap-cloud-sdk-documentation@0.1.0 prepare
> husky install

husky - Git hooks installed

added 1501 packages, and audited 1502 packages in 50s

325 packages are looking for funding
run `npm fund` for details

25 vulnerabilities (4 moderate, 21 high)

To address all issues, run:
npm audit fix

Run `npm audit` for details.
```

florian-richter reviewed (approved):
I think the `npm run style` should be part of `pre-commit` but it is also fine in `pre-push`. So merge as you see fit.
